src/main/java/org/sagebionetworks/web/client/widget/ReactComponentV2.java

```diff
@@ -181,24 +210,13 @@ public void render() {
     // This component will be rendered as a child of another React component, so destroy the root if one exists
     boolean shouldDestroyRoot = isRenderedAsReactComponentChild();
 
-    // This component may be a React child of another component, so retrieve the root widget that renders this component tree.
-    ReactComponentV2<?, ?> componentToRender = getRootReactComponentWidget();
-
-    // Asynchronously schedule root operations in case the component is in the middle of an asynchronous render cycle
-    // See https://stackoverflow.com/questions/73459382
-    Timer t = new Timer() {
-      @Override
-      public void run() {
-        if (shouldDestroyRoot) {
-          destroyRoot();
-        }
+    // Schedule destroying the root, if necessary
+    if (shouldDestroyRoot) {
+      destroyRoot();
+    }
 
-        // Create a fresh ReactElement tree and render it
-        componentToRender.createRoot();
-        componentToRender.root.render(componentToRender.createReactElement());
-      }
-    };
-    t.schedule(0);
+    // Schedule rendering
+    createRootAndRender();
```

Comment:
Note that even through the timers were changed so `destroyRoot` and `createRootAndRender` now trigger two timers, they end up on the same queue, so they will execute in order.